sys/fuchsia/fidl_mappings.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2017 syzkaller project authors. All rights reserved.
+// Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+package fuchsia
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+)
+
+// layer indicates at which layer a FidlLibrary lives in the Fuchsia build
+// system.
+type layer int
+
+const (
+	_ layer = iota
+	zircon
+	garnet
+)
+
+// FidlLibrary describes a FIDL library. It captures required details such as
+// build location, header generation, etc.
+type FidlLibrary struct {
+	layer
```

Comment:
I think it's better to use a named variable. Embedding serves very special needs and is relatively rarely used. It does not seem that we use anything that embedding gives us here: you refer to it as `.layer` throughout and it does not have any methods.